app/grandchallenge/participants/views.py

```diff
@@ -115,3 +129,103 @@ def get_success_url(self):
             "participants:registration-list",
             kwargs={"challenge_short_name": self.object.challenge.short_name},
         )
+
+
+class RegistrationQuestionList(
+    LoginRequiredMixin, PermissionListMixin, ListView
+):
+    model = RegistrationQuestion
+    permission_required = "participants.view_registrationquestion"
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        queryset = queryset.filter(challenge=self.request.challenge)
+        return queryset
+
+    def dispatch(self, request, *args, **kwargs):
+
+        if not self.request.user.has_perm(
+            "challenges.change_challenge",
+            self.request.challenge,
+        ):
+            # View contains admin side-panel
+            raise PermissionDenied
+
+        return super().dispatch(request, *args, **kwargs)
```

Comment:
I think you're getting into a bit of a mess here. Overriding the dispatch method is dangerous, especially with so many mixins. If there is something in the page that requires a different permission then that should be checked in the template.

Reply:
I agree with it being a weird mixin versus custom dispatch overriding. I was hesitant to push it into the template because it broadens the impact of this PR. In short, the challenge settings sidebar will not show unless someone has `"change_challenge"` permissions (see the challenge_settings_base.html template).

If the user `readerstudy` looks at the page `http://demo.gc.localhost:8000/participants/registration/questions/` they will see the following:
![image](https://github.com/user-attachments/assets/6f507dfb-68d6-4a9b-943a-c56f572b6d26)

Reply:
That looks fine, so long as the items are filtered. Probably we should solve that they can load the page in the first place systematically for all of the list views but would require double permission handling => another issue.

Reply:
I can imagine that if we add more subtle permissions, sections of the settings would be hidden. Indeed another issue. First this pitch PR!